adapters/criteo/models.go

```diff
@@ -0,0 +1,333 @@
+package criteo
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/mxmCherry/openrtb/v14/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type criteoRequest struct {
+	ID          string                   `json:"id,omitempty"`
+	Publisher   criteoPublisher          `json:"publisher,omitempty"`
+	User        criteoUser               `json:"user,omitempty"`
+	GdprConsent criteoGdprConsent        `json:"gdprconsent,omitempty"`
+	Slots       []criteoRequestSlot      `json:"slots,omitempty"`
+	Eids        []openrtb_ext.ExtUserEid `json:"eids,omitempty"`
+}
+
+func newCriteoRequest(slotIDGenerator slotIDGenerator, request *openrtb2.BidRequest) (criteoRequest, []error) {
+	var errs []error
+
+	// request cannot be nil by design
+
+	criteoRequest := criteoRequest{}
+
+	if request.ID != "" {
+		criteoRequest.ID = request.ID
+	}
+
+	// Extracting request slots
+	if len(request.Imp) > 0 {
+		criteoSlots, slotsErr := newCriteoRequestSlots(slotIDGenerator, request.Imp)
+		if len(slotsErr) > 0 {
+			return criteoRequest, append(errs, slotsErr...)
```

Comment:
There's no need to append `slotsErr` to `errs` here. You can return `slotsErr` directly.

Reply:
Ack